lib/web_ui/lib/src/engine/surface/clip.dart

```diff
@@ -325,16 +325,16 @@ class PersistedPhysicalShape extends PersistedContainerSurface
     }
     if (oldSurface.path != path) {
       oldSurface._clipElement?.remove();
-      // Reset style on prior element since we may have switched between
-      // rect/rrect and arbitrary path.
-      final html.CssStyleDeclaration style = rootElement.style;
-      style.transform = '';
-      style.left = '';
-      style.top = '';
-      style.borderRadius = '';
       domRenderer.setElementStyle(rootElement, 'clip-path', '');
       domRenderer.setElementStyle(rootElement, '-webkit-clip-path', '');
       _applyShape();
+      if (path == null) {
```

Comment:
The `path == null` check already exists inside the `_applyShape`. I think the clean-up should be there as well. Otherwise the clean-up won't happen along the other code paths.

Reply:
I might be missing something. The only entrypoints are apply when building for the first time and update when reusing. On first build, we just simply don't do anything for path=null. On updates, we would always hit the cleanup code path?

Reply:
Oh, I see. It's totally possible that we only need to clean up in the update when we're reusing an existing element. But I think this breaks separation of concerns. `_applyShape`'s primary purpose is to populate left/top/borderRadius according to the value of `path`. So it's not clear why it's not applying the `null` shape case.

I'm not against doing it this way, if there's a good reason for it. Maybe leave a comment explaining why the `null` case has to be outside `_applyShape`?

Reply:
Done. added comment.